pallets/runtime/tests/src/corporate_actions_test.rs

```diff
@@ -564,3 +632,887 @@ fn link_ca_docs_works() {
         link_ok(vec![id0]);
     });
 }
+
+#[test]
+fn remove_ca_works() {
+    test(|ticker, [owner, ..]| {
+        set_schedule_complexity();
+
+        let ca = |kind, rd| moment_ca(owner, ticker, kind, rd).unwrap();
+        let remove = |id| CA::remove_ca(owner.signer(), id);
+
+        let assert_no_ca = |id: CAId| {
+            assert_eq!(None, CA::corporate_actions(ticker, id.local_id));
+            assert_eq!(CA::ca_doc_link(id), vec![]);
+        };
+
+        // Remove a CA that doesn't exist, and ensure failure.
+        let id = next_ca_id(ticker);
+        assert_noop!(remove(id), Error::NoSuchCA);
+
+        // Create a CA, remove it, and ensure its no longer there.
+        ca(CAKind::Other, None);
+        add_doc(owner, ticker);
+        let docs = vec![DocumentId(0)];
+        assert_ok!(CA::link_ca_doc(owner.signer(), id, docs.clone()));
+        assert_eq!(docs, CA::ca_doc_link(id));
+        assert_ok!(remove(id));
+        assert_no_ca(id);
+
+        // Create a ballot CA, which hasn't started.
+        let time = BallotTimeRange {
```

Comment:
The constant defined earlier can be reused

Reply:
The times in that constant are wrong. The constant is basically for "we don't really care", but here the times matter and we test against them.

Reply:
Oh wait, no, they are actually the same. But alas, I would prefer to keep the times here, to clarify the test semantics, cause of the "don't care" usage of the constant.